src/backend/routers/agent.py

```diff
@@ -476,6 +511,62 @@ async def delete_agent_tool_metadata(
     return DeleteAgentToolMetadata()
 
 
+@router.post("/batch_upload_files", response_model=list[UploadFileResponse])
```

Comment:
I decided to separate upload into two endpoints.

One used in a conversation, this one for the agent creation screen.

Rationale for this is that depending which kind of upload we do there are side effects. For conversation file upload we need to create conversations if they don't exists. Here we don't need to do that.

Also the semantics of `/conversation/upload` doesn't really make sense if we are uploading for an agent.